2.12alpha16/17 crash in test lab pre-launch report

```
FATAL EXCEPTION: main
Process: com.ichi2.anki, PID: 31772
java.lang.RuntimeException: Unable to pause activity {com.ichi2.anki/com.ichi2.anki.NoteEditor}: java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
	at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:4124)
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4556)
	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4621)
	at android.app.ActivityThread.-wrap5(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1757)
	at android.os.Handler.dispatchMessage(Handler.java:105)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6938)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
Caused by: java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
	at androidx.fragment.app.FragmentManagerImpl.checkStateLoss(FragmentManagerImpl.java:2)
	at androidx.fragment.app.FragmentManagerImpl.enqueueAction(FragmentManagerImpl.java:1)
	at androidx.fragment.app.FragmentManagerImpl.popBackStack(FragmentManagerImpl.java:2)
	at com.ichi2.anki.AnkiActivity.dismissAllDialogFragments(AnkiActivity.java:1)
	at com.ichi2.anki.NoteEditor.onPause(NoteEditor.java:1)
	at android.app.Activity.performPause(Activity.java:7391)
	at android.app.Instrumentation.callActivityOnPause(Instrumentation.java:1414)
	at androidx.test.runner.MonitoringInstrumentation.callActivityOnPause(MonitoringInstrumentation.java:17)
	at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:4113)
	... 10 more
```

Device logcat - [logcat.txt](https://github.com/ankidroid/Anki-Android/files/5099833/logcat.txt)


Device screen recording - https://youtu.be/XrEPQDy395c

--- a/AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java
@@ -393,7 +393,13 @@ public class NoteEditor extends AnkiActivity {
     }
 
     @Override
-    protected void onSaveInstanceState(Bundle savedInstanceState) {
+    protected void onSaveInstanceState(@NonNull Bundle savedInstanceState) {
+        addInstanceStateToBundle(savedInstanceState);
+        super.onSaveInstanceState(savedInstanceState);
+    }
+
+
+    private void addInstanceStateToBundle(@NonNull Bundle savedInstanceState) {
         Timber.i("Saving instance");
         savedInstanceState.putInt("caller", mCaller);
         savedInstanceState.putBoolean("addNote", mAddNote);
@@ -405,7 +411,6 @@ public class NoteEditor extends AnkiActivity {
         }
         savedInstanceState.putStringArrayList("tags", mSelectedTags);
         savedInstanceState.putBundle("editFields", getFieldsAsBundle(false));
-        super.onSaveInstanceState(savedInstanceState);
     }
 
 
@@ -1000,17 +1005,7 @@ public class NoteEditor extends AnkiActivity {
 
             case R.id.action_preview:
                 Timber.i("NoteEditor:: Preview button pressed");
-                Intent previewer = new Intent(NoteEditor.this, CardTemplatePreviewer.class);
-
-                previewer.putExtra("ordinal", 0);
-                previewer.putExtra(TemporaryModel.INTENT_MODEL_FILENAME, TemporaryModel.saveTempModel(this, mEditorNote.model()));
-
-                // Send the previewer all our current editing information
-                Bundle noteEditorBundle = new Bundle();
-                onSaveInstanceState(noteEditorBundle);
-                noteEditorBundle.putBundle("editFields", getFieldsAsBundle(true));
-                previewer.putExtra("noteEditorBundle", noteEditorBundle);
-                startActivityForResultWithoutAnimation(previewer, REQUEST_PREVIEW);
+                performPreview();
                 return true;
 
             case R.id.action_save:
@@ -1061,6 +1056,22 @@ public class NoteEditor extends AnkiActivity {
     // ----------------------------------------------------------------------------
     // CUSTOM METHODS
     // ----------------------------------------------------------------------------
+
+    @VisibleForTesting
+    void performPreview() {
+        Intent previewer = new Intent(NoteEditor.this, CardTemplatePreviewer.class);
+
+        previewer.putExtra("ordinal", 0);
+        previewer.putExtra(TemporaryModel.INTENT_MODEL_FILENAME, TemporaryModel.saveTempModel(this, mEditorNote.model()));
+
+        // Send the previewer all our current editing information
+        Bundle noteEditorBundle = new Bundle();
+        addInstanceStateToBundle(noteEditorBundle);
+        noteEditorBundle.putBundle("editFields", getFieldsAsBundle(true));
+        previewer.putExtra("noteEditorBundle", noteEditorBundle);
+        startActivityForResultWithoutAnimation(previewer, REQUEST_PREVIEW);
+    }
+
 
     /**
      * finish when sd card is ejected

--- a/AnkiDroid/src/test/java/com/ichi2/anki/NoteEditorTest.java
+++ b/AnkiDroid/src/test/java/com/ichi2/anki/NoteEditorTest.java
@@ -27,7 +27,6 @@ import com.ichi2.anki.multimediacard.fields.IField;
 import com.ichi2.libanki.Consts;
 import com.ichi2.libanki.Model;
 import com.ichi2.libanki.Note;
-import com.ichi2.utils.JSONObject;
 
 import org.junit.Ignore;
 import org.junit.Test;
@@ -45,6 +44,7 @@ import androidx.test.core.app.ActivityScenario;
 import androidx.test.ext.junit.runners.AndroidJUnit4;
 
 import static com.ichi2.compat.Compat.EXTRA_PROCESS_TEXT;
+import static com.ichi2.testutils.AnkiAssert.assertDoesNotThrow;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.contains;
 import static org.hamcrest.Matchers.is;
@@ -270,6 +270,13 @@ public class NoteEditorTest extends RobolectricTest {
         i.putExtra(EXTRA_PROCESS_TEXT, "hello\nworld");
         NoteEditor editor = startActivityNormallyOpenCollectionWithIntent(NoteEditor.class, i);
         assertThat(Arrays.asList(editor.getCurrentFieldStrings()), contains("hello\nworld", ""));
+    }
+
+    @Test
+    public void previewWorksWithNoError() {
+        // #6923 regression test - Low value - Could not make this fail as onSaveInstanceState did not crash under Robolectric.
+        NoteEditor editor = getNoteEditorAddingNote(FromScreen.DECK_LIST, NoteEditor.class);
+        assertDoesNotThrow(editor::performPreview);
     }
 
     private Intent getCopyNoteIntent(NoteEditor editor) {

